packages/node-sdk/paima-utils-backend/src/cde-access-internals.ts

```diff
@@ -182,11 +183,30 @@ export async function internalGetAllOwnedErc6551Accounts(
 export async function internalGetCardanoAddressDelegation(
   readonlyDBConn: Pool,
   address: string
-): Promise<string | null> {
+): Promise<{ events: { pool: string | null; epoch: number }[]; currentEpoch: number } | null> {
   const results = await cdeCardanoPoolGetAddressDelegation.run({ address }, readonlyDBConn);
   if (results.length === 0) {
     return null;
   }
 
-  return results[0].pool;
+  const currentEpoch = await getCardanoEpoch.run(undefined, readonlyDBConn);
+
+  if (currentEpoch.length === 0) {
+    throw new Error('Current epoch table not initialized');
+  }
+
+  if (currentEpoch[0].epoch === results[results.length - 1].epoch) {
```

Comment:
Are you sure this `if` condition shouldn't be flipped?

I think probably we don't even need this if/else. We can use the same code and then just use `Math.min(0, results.length - 1)` for the index

Reply:
Flipped? Like negating the condition?

This is checking if one of the entries is for the current epoch, in which case all the entries are returned (there can be 2 max), so in this branch `events` is the entire list.

In the other branch instead there is only one result max, because both have to be `< currentEpoch`.

We could simplify the logic by just always returning all of the entries though. I just thought it was leaking implementation details.

Also this `Math.min(0, results.length - 1)` would be always 0, since there is already an early return for the case when there are no delegations, so I'm not sure what you mean.

Reply:
Okay, I think I misunderstood the code originally

We should have a comment block that explains what is going on here 

something like
```js
/**
 * If the most recent delegation is the current epoch, we need to return the list of recent delegations so the app can know what delegation the user had beforehand since delegations only matters once they cross an epoch boundary
 * If the most recent delegation isn't from the current epoch, we know it's the one that is active now
*/
```

either that, or we explain this in the Paima Engine docs and then put a link to the docs as the comment (to avoid the same explanation being in both places)

Reply:
I did put an explanation of the result at https://github.com/PaimaStudios/paima-engine-docs/pull/19 

So I could link to that. Although I would be a bit concerned about the domain/url changing if I link to the public version, so I feel like it's better to just replicate it here, maybe a shorter version.